client/src/ingredients/browser-compatibility-table/index.test.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { BrowserCompatibilityTable } from "./index.js";
+
+/**
+ * Based on `stumptown/packaged/html/reference/elements/abbr.json`
+ */
+const data = {
+  query: "html.elements.abbr",
+  data: {
+    __compat: {
+      mdn_url: "https://developer.mozilla.org/docs/Web/HTML/Element/abbr",
+      support: {
+        chrome: {
+          version_added: "2"
+        },
+        chrome_android: {
+          version_added: true
+        },
+        edge: {
+          version_added: true
+        },
+        firefox: {
+          version_added: "1",
+          notes:
+            "Before Firefox 4, this element implemented the <code>HTMLSpanElement</code> interface instead of the standard <code>HTMLElement</code> interface."
+        },
+        firefox_android: {
+          version_added: "4"
+        },
+        ie: {
+          version_added: "7"
+        },
+        opera: {
+          version_added: true
+        },
+        opera_android: {
+          version_added: true
+        },
+        safari: {
+          version_added: true
+        },
+        safari_ios: {
+          version_added: true
+        },
+        samsunginternet_android: {
+          version_added: true
+        },
+        webview_android: {
+          version_added: true
+        }
+      },
+      status: {
+        experimental: false,
+        standard_track: true,
+        deprecated: false
+      }
+    }
+  }
+};
+
+it("renders without crashing", () => {
+  const { container } = render(<BrowserCompatibilityTable data={data} />);
+  expect(container).toMatchSnapshot();

```

Comment:
On other note, what do you feel about this snapshot test case? It's a lazy way to write test case 😂 

But it may fail every time:

1. stumptown content schema changes, --> one need to manually update the test case input
2. BrowserCompatibilityTable component structure changes --> one can just easily update the snapshot

On one hand, it makes it very explicit when thing changes, but on the other hand, it may be failing often if the project isn't stabilized yet

Reply:
Imo that's fine or even a good thing, for the reasons you outlined. And since jest makes updating snapshots trivial (while making it very clear what changed), this is in my view a non-problem.

Reply:
I'm personally very negative towards snapshot testing. It's a test of the tests and not of the API. So many times, I've seen people blindly just update the snapshots if the tests don't pass. It also causes massive files to be checked into git that aren't human-generated. 

Sure, there's a chance that a human might notice that the snapshot tests suddenly broke (for a real reason), but what are the chances that that human genuinely starts to investigate what the difference is and how it relates to his/her change of the functionality? 

Also, there's something about the snapshot files that they are is closer to "testing the implementation details" rather than "testing the API" and that's that the snapshot includes a bunch of stuff that isn't relevant but might break. Suppose we decide to change a certain icon. So from `<i class="undefined"/>` to `<i class="fa-unknown"/>` that would break the tests. But it's not fair because the functionality hasn't changed. 

To me, snapshot testing is like testing the content of a `.css` file instead of looking at the visual difference it presents when used.

Reply:
> I'm personally very negative towards snapshot testing. It's a test of the tests and not of the API. So many times, I've seen people blindly just update the snapshots if the tests don't pass. It also causes massive files to be checked into git that aren't human-generated.

I do think this is a problem that should be caught in code-review, i.e. when reviewing PRs we should make sure to also check snapshot diffs (which imo are quite readable, even when large). I agree with the general point about them potentially being massive, I think reading the diff of a snapshot tends to be more interesting than their whole contents.

> Also, there's something about the snapshot files that they are is closer to "testing the implementation details" rather than "testing the API" and that's that the snapshot includes a bunch of stuff that isn't relevant but might break. Suppose we decide to change a certain icon. So from `<i class="undefined"/>` to `<i class="fa-unknown"/>` that would break the tests. But it's not fair because the functionality hasn't changed.

Imo snapshot tests are closer to testing the API the user interacts with, i.e. the UI. So with that view we are changing the interface people interact with, hence the tests breaking is what we'd want. But I see your general point, and we've talked about that before, that it is a little bit like armor, we're better protected but we move slower.

In my experience without Snapshot testing, UI testing just becomes a more manual less exhaustive version of snapshot testing. My vote is for trying to keep it in but be cautious of whether we're actually properly using it when changing things and reviewing PRs.

Reply:
I do agree that snapshot testing of a big component isn't helpful at all, since the developer will just "blindly update snapshot", and since it's a big diff, the code reviewer won't even bother to look at it.

> the snapshot files that they are is closer to "testing the implementation details" rather than "testing the API"

I also agree on this point, since snapshot testing captures all every single details that may result in test failing all the time when anything changes.

I'm not very familiar with best practices in (end-to-end) UI testing, but I believe, these kind of tests generally asserts if some elements are present and acts on certain elements to see the effects. In contrast, snapshot testing of a big component will do a snapshot of all elements and does not test any interaction.

With regard to this PR, I'll be removing the snapshot test for this file first. I think further discussion on what to test and how to test could be tracked in a separate issue.

✌️